src/main/java/org/checkerframework/checker/noliteral/qual/MaybeConstant.java

```diff
@@ -0,0 +1,36 @@
+package org.checkerframework.checker.noliteral.qual;
+
+import java.lang.annotation.ElementType;
+import java.lang.annotation.Retention;
+import java.lang.annotation.RetentionPolicy;
+import java.lang.annotation.Target;
+import org.checkerframework.framework.qual.DefaultInUncheckedCodeFor;
+import org.checkerframework.framework.qual.LiteralKind;
+import org.checkerframework.framework.qual.QualifierForLiterals;
+import org.checkerframework.framework.qual.SubtypeOf;
+import org.checkerframework.framework.qual.TypeUseLocation;
+
+/**
+ * The top type for the NoLiteral Checker's type system.
+ *
+ * <p>This type represents an expression that might contain a literal, like 5 or {0xa, 0xb}. It is
+ * inferred for all manifest literals in the program.
+ *
+ * <p>This is also the type of any expression that contains a subexpression that may be a constant.
```

Comment:
The Javadoc for one of the annotations is an unexpected place to find the expression typing rules.  They belong in an overview document, not here.